### PR TITLE
fix(frames): remove non absolute non url frame.abs_path coming from default stack parser

### PR DESCRIPTION
**Pull request by krystofwoldrich**
## :loudspeaker: Type of change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring

## :scroll: Description

In JS SDK `frame.abs_path` was added and that caused the `from.abs_path` to be relative. This lead to source maps failure in the product.

related changes in JS getsentry/sentry-javascript#7167

## :bulb: Motivation and Context

fixes: https://github.com/getsentry/sentry-react-native/issues/2879

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled
- [x] All tests passing
- [x] No breaking changes

## :crystal_ball: Next steps
- [x] Add test so this won't happen again

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+### Fixes
+
+- Remove non URL `frame.abs_path` which was causing source maps to fail ([#2876](https://github.com/getsentry/sentry-react-native/pull/2876))
```
Comment (krystofwoldrich) [inline CHANGELOG.md:7]:
```suggestion
- Remove non URL `frame.abs_path` which was causing source maps to fail ([#2891](https://github.com/getsentry/sentry-react-native/pull/2891))
```

File: src/js/sdk.tsx [hunk 2]
```diff
@@ -126,6 +126,7 @@ export function init(passedOptions: ReactNativeOptions): void {
                 ? `${appPrefix}${frame.filename}`
                 : `${appPrefix}/${frame.filename}`;
           }
+          delete frame.abs_path;
```
Comment (marandaneto) [inline src/js/sdk.tsx:129]: M: `abs_path` is part of the protocol, so symbolication should work anyway, for fixing this issue is fine as of now, but I wonder if we should not fix ingestion when both `filename` and `abs_path` is provided? can you double check with the stack trace team?

**Review by marandaneto — APPROVED** [review]

File: src/js/sdk.tsx (hunk 2 above)
Comment (krystofwoldrich) [reply src/js/sdk.tsx:129]: [Relay does exactly that](https://github.com/getsentry/relay/blob/1e470ec87aee8fb9214844a1cf5eaa5257ea3d69/relay-general/src/store/normalize/stacktrace.rs#L27-L29) it copies it from filename, so until we need the two fields in the SDK I think we can save a bit the envelope size and send only the filename.

Comment (krystofwoldrich) [reply src/js/sdk.tsx:129]: The reason why it didn't work is becuae the default frame parse now copies the name before the changes needed for RN. 

I've talked to the web team and I still don't exactly understand what they use it for, but we can adjust that later.

Comment (krystofwoldrich) [reply src/js/sdk.tsx]: So it's used for identifying new Debug Meta Image Type Source Maps, I will have to talk more to the web team on how it works.

https://develop.sentry.dev/sdk/event-payloads/debugmeta/#source-map-images

**Comment by krystofwoldrich** [comment]
@marandaneto I've added tests for the rewrite frames. Could you review it too?

File: src/js/sdk.tsx (hunk 2 above)
Comment (marandaneto) [reply src/js/sdk.tsx]: alright, let's create a repo maintenance issue for that, so we don't forget it.

**Review by marandaneto — APPROVED** [review]
I liked the approach of splitting up the `RewriteFrames` and making it testable + adding unit tests for multiple cases, thank you!

File: src/js/sdk.tsx (hunk 2 above)
Comment (krystofwoldrich) [reply src/js/sdk.tsx]: https://github.com/getsentry/sentry-react-native/issues/2892

**Pull request merged**